app/src/main/java/com/sjh/autosummary/core/domain/UpdateChatSummaryUseCase.kt

```diff
@@ -0,0 +1,130 @@
+package com.sjh.autosummary.core.domain
+
+import android.util.Log
+import com.sjh.autosummary.core.data.model.ChatRequest
+import com.sjh.autosummary.core.data.repository.ChatRepository
+import com.sjh.autosummary.core.data.repository.SummaryRepository
+import com.sjh.autosummary.core.model.ChatRoleType
+import com.sjh.autosummary.core.model.ChatSummary
+import com.sjh.autosummary.core.model.MessageContent
+import kotlinx.coroutines.Dispatchers
+import kotlinx.coroutines.withContext
+import kotlinx.serialization.encodeToString
+import kotlinx.serialization.json.Json
+import javax.inject.Inject
+
+class UpdateChatSummaryUseCase @Inject constructor(
+    private val chatRepository: ChatRepository,
+    private val summaryRepository: SummaryRepository,
+) {
+    suspend operator fun invoke(messageContent: MessageContent): Result<Boolean> =
+        withContext(Dispatchers.IO) {
+            try {
+                // 1. 저장된 모든 요약 정보 가져오기
+                val retrieveResult = summaryRepository.retrieveAllChatSummaries()
+                // (요약 불가 처리)
+                if (retrieveResult.isFailure) return@withContext Result.success(false)
+
+                val summaries = retrieveResult.getOrThrow()
+                val summariesInJson = convertChatSummaryToJson(summaries.firstOrNull())
+                Log.d("whatisthis", "summariesInJson $summariesInJson")
+
+                // 2. 답변 내용 요약
+                val summarizedResponseResult = chatRepository.createChatCompletion(
+                    chatRequest = ChatRequest(
+                        requestMessage = messageContent
+                    )
+                )
+                if (summarizedResponseResult.isFailure) return@withContext Result.success(false)
+                summarizedResponseResult.getOrThrow().responseMessage?.let { content ->
+                    // 3. 요약된 답변 내용과 모든 요약 정보를 합쳐 요청 메시지 생성
+                    val summaryRequestContent =
+                        buildRequestFormForSummary(summariesInJson, content.content)
```

Comment:
Suggestion: Simple is the best. Lambda is not simple for readers but simple only for you.
```kotlin
val responseMessage = summarizedResponseResult.getOrThrow().responseMessage
if (responseMessage == null) {
  return Result.success(true)
}

val summaryRequestContent = buildRequestFormForSummary(summariesInJson, responseMessage.content)
...
```